api/src/main/java/com/epam/pipeline/manager/pipeline/documents/templates/VersionStorageReportTemplateManager.java

```diff
@@ -0,0 +1,339 @@
+        fillTemplate(report, loaded, gitDiff, reportFilters);
+        results.add(Pair.of(SUMMARY + NAME_SEPARATOR + loaded.getName() + NAME_SEPARATOR +
+                ReportDataExtractor.DATE_FORMAT.format(DateUtils.now()) + DOCX, report));
+        if (reportFilters.isArchive()) {
+            results.addAll(
+                    prepareDiffsForReportDoc(loaded, gitDiff,
+                        GitDiffReportFilter.builder()
+                            .groupType(reportFilters.getGroupType())
+                            .commitsFilter(reportFilters.getCommitsFilter())
+                            .includeDiff(reportFilters.isIncludeDiff())
+                            .build()
+                    )
+            );
+        }
+        return results;
+    }
+
+    private List<Pair<String, XWPFDocument>> prepareDiffsForReportDoc(final Pipeline loaded, final GitDiff gitDiff,
+                                                                      final GitDiffReportFilter reportFilters) {
+        return CommitDiffsGrouping.builder()
+                .includeDiff(reportFilters.isIncludeDiff())
+                .diffGrouping(
+                        getGroupType(reportFilters) == CommitDiffsGrouping.GroupType.BY_COMMIT
+                            ? gitDiff.getEntries().stream()
+                            .collect(Collectors.groupingBy(e -> e.getCommit().getCommit()))
+                            : gitDiff.getEntries().stream()
+                            .collect(Collectors.groupingBy(e -> e.getDiff().getToFileName()))
+                ).build().getDiffGrouping()
+                .entrySet()
+                .stream()
+                .map(e -> Pair.of(e.getKey(), e.getValue()))
+                .map(pair ->
+                        Pair.of(
+                            pair.getFirst(),
+                            GitDiff.builder()
+                                .entries(pair.getSecond())
+                                .filters(gitDiff.getFilters())
+                                .build()
+                        )
+                ).map(p -> {
+                    try {
+                        final XWPFDocument report = new XWPFDocument(
+                                new FileInputStream(getVersionStorageTemplatePath())
+                        );
+                        int toDelete = 0;
+                        while (report.getBodyElements().size() != 1) {
+                            IBodyElement element = report.getBodyElements().get(toDelete);
+                            if (element.getElementType() == BodyElementType.PARAGRAPH &&
+                                    ((XWPFParagraph) element).getText()
+                                            .contains(VSReportTemplates.COMMIT_DIFFS.template)) {
+                                toDelete += 1;
+                            }
+                            report.removeBodyElement(toDelete);
+                        }
```

Comment:
Not exactly
In order to not to introduce new report template, we clean up existing on and skip only paragraph where `commit_diffs` template exists
regarding to `removeBodyElement` there shouldn't be any unexpected exceptions such as NPE or IndexOutOfBoundException because this method check that `toDelete` index is in bounds of bodyElements array